package effect.protocol;

message EffectSetting
{
	enum Type {
		Unknown = 0;
		Plexus = 1;
	};

	// Identify the effect either by type or by class
	optional Type type = 1;
	optional string effect_class = 2;

	// If id is 0, one will be assinged
	optional uint32 id = 3;
	optional string name = 4;

	optional uint32 start_time = 5;
	optional uint32 end_time = 6;

	// Config, either as a encoded protobuf, or as a filename containing the config
	optional bytes config_msg = 7;
	optional string config_file = 8;
}

message EffectSettings
{
	repeated EffectSetting effect_setting = 1;
	optional string soundtrack = 2;
}
